actors/market/tests/harness.rs

```diff
@@ -0,0 +1,750 @@
+use cid::Cid;
+use num_traits::FromPrimitive;
+use std::collections::HashMap;
+
+use fil_actor_market::{
+    balance_table::BalanceTable, ext, ext::miner::GetControlAddressesReturnParams,
+    gen_rand_next_epoch, ActivateDealsParams, Actor as MarketActor, ClientDealProposal, DealArray,
+    DealMetaArray, DealProposal, DealState, Method, OnMinerSectorsTerminateParams,
+    PublishStorageDealsParams, PublishStorageDealsReturn, State, WithdrawBalanceParams,
+    WithdrawBalanceReturn, PROPOSALS_AMT_BITWIDTH,
+};
+use fil_actor_power::{CurrentTotalPowerReturn, Method as PowerMethod};
+use fil_actor_reward::Method as RewardMethod;
+use fil_actor_verifreg::UseBytesParams;
+use fil_actors_runtime::runtime::{Policy, Runtime};
+use fil_actors_runtime::{
+    test_utils::*, ActorError, SetMultimap, BURNT_FUNDS_ACTOR_ADDR, CRON_ACTOR_ADDR,
+    REWARD_ACTOR_ADDR, STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR, SYSTEM_ACTOR_ADDR,
+    VERIFIED_REGISTRY_ACTOR_ADDR,
+};
+use fvm_ipld_encoding::{to_vec, RawBytes};
+use fvm_shared::bigint::BigInt;
+use fvm_shared::clock::{ChainEpoch, EPOCH_UNDEFINED};
+use fvm_shared::crypto::signature::Signature;
+use fvm_shared::deal::DealID;
+use fvm_shared::piece::PaddedPieceSize;
+use fvm_shared::reward::ThisEpochRewardReturn;
+use fvm_shared::sector::StoragePower;
+use fvm_shared::smooth::FilterEstimate;
+use fvm_shared::{
+    address::Address, econ::TokenAmount, error::ExitCode, METHOD_CONSTRUCTOR, METHOD_SEND,
+};
+
+// Define common set of actor ids that will be used across all tests.
+pub const OWNER_ID: u64 = 101;
+pub const PROVIDER_ID: u64 = 102;
+pub const WORKER_ID: u64 = 103;
+pub const CLIENT_ID: u64 = 104;
+pub const CONTROL_ID: u64 = 200;
+
+pub const OWNER_ADDR: Address = Address::new_id(OWNER_ID);
+pub const PROVIDER_ADDR: Address = Address::new_id(PROVIDER_ID);
+pub const WORKER_ADDR: Address = Address::new_id(WORKER_ID);
+pub const CLIENT_ADDR: Address = Address::new_id(CLIENT_ID);
+pub const CONTROL_ADDR: Address = Address::new_id(CONTROL_ID);
+
+pub struct MinerAddresses {
+    pub owner: Address,
+    pub worker: Address,
+    pub provider: Address,
+    pub control: Vec<Address>,
+}
+
+// Use the predefined actor addresses by default
+impl Default for MinerAddresses {
+    fn default() -> Self {
+        MinerAddresses {
+            owner: OWNER_ADDR,
+            worker: WORKER_ADDR,
+            provider: PROVIDER_ADDR,
+            control: vec![CONTROL_ADDR],
+        }
+    }
+}
+
+pub fn setup() -> MockRuntime {
+    let mut actor_code_cids = HashMap::default();
```

Comment:
we could initialize this hashmap from an array to avoid mutability and number of calls (also, code golf)

Reply:
Fixed.